gravitee-am-identityprovider/gravitee-am-identityprovider-mongo/src/main/java/io/gravitee/am/identityprovider/mongo/user/MongoUserProvider.java

```diff
@@ -274,4 +283,29 @@ private byte[] createSalt() {
         random.nextBytes(salt);
         return salt;
     }
+
+    /**
+     * Create or update index on username field
+     */
+    private void createOrUpdateIndex() {
+        if (ensureIndexOnStart) {
+            getUserNameIndex()
+                    .andThen(Completable.fromPublisher(usersCollection.createIndex(new Document(configuration.getUsernameField(), 1), new IndexOptions().unique(true))))
+                    .subscribe(
+                            () -> LOGGER.debug("Unique index on username created"),
+                            err -> LOGGER.error("An error occurred while creating index {} with unique constraints", INDEX_USERNAME, err));
+        }
+    }
+
+    private Completable getUserNameIndex() {
+        return Observable.fromPublisher(usersCollection.listIndexes())
+                .map(document -> document.getString("name"))
+                .flatMapCompletable(indexName -> {
+                    if (indexName.equals(INDEX_USERNAME)) {
+                        return Completable.fromPublisher(usersCollection.dropIndex(indexName));
+                    } else {
+                        return Completable.complete();
+                    }
+                });
+    }
```

Comment:
Remark: If you create a index with the same name than the previous one (username_1), then the index will be deleted and created after each startup. An index name like `u_1` should be define here https://github.com/gravitee-io/gravitee-access-management/pull/2874/files#diff-5dca5bac1e1df3609de0a22941dcfc0ee1289e0150b95be9f53634d3abda2adcR293

Reply:
tested and you are correct, introduced `u1_unique` index name.